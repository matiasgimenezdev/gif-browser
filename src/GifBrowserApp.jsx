import { useState } from 'react';
import Form from './components/Form/Form';
import GifGrid from './components/GifGrid/GifGrid';

function GifExpertApp() {
	const [categories, setCategories] = useState(['Valorant']);

	const handleAdd = (category) => {
		if (!categories.includes(category)) {
			setCategories([category, ...categories]);
		}
	};

	return (
		<>
			<h1>Gif Expert Application</h1>
			<Form handleAdd={handleAdd} />

			{categories.map((category) => {
				return <GifGrid key={category} category={category} />;
			})}
		</>
	);
}

export default GifExpertApp;
