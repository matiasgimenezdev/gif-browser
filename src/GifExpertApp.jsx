import { useState } from 'react';
function GifExpertApp() {
	const [categories, setCategories] = useState([
		'Fútbol',
		'Programación',
		'Arte',
	]);
	const [category, setCategory] = useState('');

	const handleChange = (event) => {
		setCategory(event.target.value);
	};

	const handleAdd = (event) => {
		setCategories([category, ...categories]);
	};

	return (
		<>
			<h1>Gif Expert Application</h1>
			<input
				placeholder='Add category...'
				name='category'
				id='category'
				onChange={handleChange}
			/>
			<button onClick={handleAdd}>Add category</button>
			<ol>
				{categories.map((category) => {
					return <li key={category}>{category}</li>;
				})}
			</ol>
		</>
	);
}

export default GifExpertApp;
