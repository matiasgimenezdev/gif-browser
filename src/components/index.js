export * from './Form/Form';
export * from './GifGrid/GifGrid';
export * from './GifItem/GifItem';
