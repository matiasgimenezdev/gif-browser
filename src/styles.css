html {
	box-sizing: border-box;
	font-family: 'Roboto', sans-serif;
}

*,
*::after,
*::before {
	padding: 0;
	margin: 0;
	box-sizing: inherit;
}

h1 {
	padding: 0.8rem 0.5rem;
}

ol {
	padding: 1rem 2rem;
}

input {
	margin: 0.5rem;
	height: 2rem;
	outline: none;
	border: none;
	border-bottom: black solid 1px;
}
