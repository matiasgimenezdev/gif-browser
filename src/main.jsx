import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
import GifBrowserApp from './GifBrowserApp';

ReactDOM.createRoot(document.getElementById('root')).render(
	<React.StrictMode>
		<GifBrowserApp />
	</React.StrictMode>
);
