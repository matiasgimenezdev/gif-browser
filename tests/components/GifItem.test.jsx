import { GifItem } from './../../src/components/';
import { render, screen } from '@testing-library/react';

describe('Pruebas en <GifItem>', () => {
	const title = 'Messi';
	const url = 'https://fifaworldcup.com/messi.jpg';

	test('<GifItem> debe matchear con el snapshot', () => {
		const { container } = render(<GifItem title={title} url={url} />);
		expect(container).toMatchSnapshot();
	});

	test('Debe renderizar la imagen con la url y el ALT indicados ', () => {
		render(<GifItem title={title} url={url} />);
		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('Debe renderizar el title ', () => {
		render(<GifItem title={title} url={url} />);
		expect(screen.getByText(title)).toBeTruthy();
	});
});
