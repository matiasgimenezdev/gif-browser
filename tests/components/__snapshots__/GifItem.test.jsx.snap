// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem> <GifItem> debe matchear con el snapshot 1`] = `
<div>
  <div
    class="gifCard"
  >
    <img
      alt="Messi"
      src="https://fifaworldcup.com/messi.jpg"
    />
    <figcaption>
      Messi
    </figcaption>
  </div>
</div>
`;
